### PR TITLE
[file-system] Remove deprecated UploadProgressData totalByteSent field

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-7226

Follow up PR updating docs -> #22278

# How

This PR removes the deprecated `totalByteSent` field from UploadProgressData

# Test Plan

Run NCL

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-file-system/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+- Removed the deprecated totalByteSent field from UploadProgressData. ([#22277](https://github.com/expo/expo/pull/22277) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```
Comment (Kudo) [inline packages/expo-file-system/CHANGELOG.md:7]:
```suggestion
- Removed the deprecated `UploadProgressData.totalByteSent` field. ([#22277](https://github.com/expo/expo/pull/22277) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```

File: packages/expo-file-system/src/FileSystem.types.ts [hunk 2]
```diff
@@ -199,40 +195,40 @@ export type FileInfo =
...
-    /**
-     * The last modification time of the file expressed in seconds since epoch.
-     */
-    modificationTime: number;
-    /**
-     * Present if the `md5` option was truthy. Contains the MD5 hash of the file.
-     */
-    md5?: string;
-  } |
+  | {
```
Comment (Kudo) [inline packages/expo-file-system/src/FileSystem.types.ts:198]: is the `| { ... } | { ... }` format intentional?

Comment (gabrieldonadel) [reply packages/expo-file-system/src/FileSystem.types.ts:198]: hmm seems that my linter messed this up, let me revert it

File: packages/expo-file-system/CHANGELOG.md (hunk 1 above)
Comment (gabrieldonadel) [reply packages/expo-file-system/CHANGELOG.md]: fixed

File: packages/expo-file-system/src/FileSystem.types.ts (hunk 2 above)
Comment (gabrieldonadel) [reply packages/expo-file-system/src/FileSystem.types.ts]: reverted

**Review by Kudo — APPROVED** [review]
👍

**Pull request merged**